### PR TITLE
[SPARK-34525][SQL][DOCS] Update documentation for various DDLs to reflect alternative key value notation 

### PR DESCRIPTION
**Pull request by nikriek**
### What changes were proposed in this pull request?

This change updates the documentation for several table DDLs (e.g. [Create Table DDL Datasource](https://spark.apache.org/docs/latest/sql-ref-syntax-ddl-create-table-datasource.html),  [Create Table DDL Hiveformat](https://spark.apache.org/docs/latest/sql-ref-syntax-ddl-create-table-hiveformat.html),  [Create Table DDL Like](https://spark.apache.org/docs/latest/sql-ref-syntax-ddl-create-table-like.html)) to reflect the alternative notation of key-value pairs `( key1 [=] val1, key2 [=] val2, ... )`.

### Why are the changes needed?

Documentation update to reflect alternative notation

### Does this PR introduce _any_ user-facing change?

Yes, the previous documentation did not account for the alternative notation of key value pairs.

Example for  [Create Table DDL Datasource](https://spark.apache.org/docs/latest/sql-ref-syntax-ddl-create-table-datasource.html)
Previous:
```sql
CREATE TABLE [ IF NOT EXISTS ] table_identifier
    [ ( col_name1 col_type1 [ COMMENT col_comment1 ], ... ) ]
    USING data_source
    [ OPTIONS ( key1=val1, key2=val2, ... ) ]
    [ PARTITIONED BY ( col_name1, col_name2, ... ) ]
    [ CLUSTERED BY ( col_name3, col_name4, ... ) 
        [ SORTED BY ( col_name [ ASC | DESC ], ... ) ] 
        INTO num_buckets BUCKETS ]
    [ LOCATION path ]
    [ COMMENT table_comment ]
    [ TBLPROPERTIES ( key1=val1, key2=val2, ... ) ]
    [ AS select_statement ]
```

Now:
```sql
CREATE TABLE [ IF NOT EXISTS ] table_identifier
    [ ( col_name1 col_type1 [ COMMENT col_comment1 ], ... ) ]
    USING data_source
    [ OPTIONS  ( key1 [=] val1, key2 [=] val2, ... ) ]
    [ PARTITIONED BY ( col_name1, col_name2, ... ) ]
    [ CLUSTERED BY ( col_name3, col_name4, ... ) 
        [ SORTED BY ( col_name [ ASC | DESC ], ... ) ] 
        INTO num_buckets BUCKETS ]
    [ LOCATION path ]
    [ COMMENT table_comment ]
    [ TBLPROPERTIES ( key1 [=] val1, key2 [=] val2, ... ) ]
    [ AS select_statement ]
```

### How was this patch tested?

No tests because it was just a documentation change

### CODE REVIEW

**Comment by maropu** [comment]
ok to test

**Comment by maropu** [comment]
It seems the other DDL stats have the same syntax, e.g. `ALTER TABLE`. Could you fix them, too, in this PR?

File: docs/sql-ref-syntax-ddl-create-table-datasource.md [hunk 1]
```diff
@@ -29,14 +29,14 @@ The `CREATE TABLE` statement defines a new table using a Data Source.
 CREATE TABLE [ IF NOT EXISTS ] table_identifier
     [ ( col_name1 col_type1 [ COMMENT col_comment1 ], ... ) ]
     USING data_source
-    [ OPTIONS ( key1=val1, key2=val2, ... ) ]
+    [ OPTIONS ( ( key1=val1, key2=val2, ... ) | ( key1 val1, key2 val2, ... ) ) ]
```
Comment (huaxingao) [inline docs/sql-ref-syntax-ddl-create-table-datasource.md:32]: I guess we can use `[ ]`to make the `=` optional?
```
[ OPTIONS ( key1 [ = ] val1 [  , ...  ]  )  ]
```

Comment (nikriek) [reply docs/sql-ref-syntax-ddl-create-table-datasource.md:32]: Agree, I don't have strong arguments against your version. I would argue that the my version is more readable for a documentation.

Comment (huaxingao) [reply docs/sql-ref-syntax-ddl-create-table-datasource.md:32]: I am OK either way. As long as we keep these consistent for all the ddl statements, I am fine.

Comment (nikriek) [reply docs/sql-ref-syntax-ddl-create-table-datasource.md]: I ultimately went with your proposal 👍  My version didn't feel that readable and concise anymore when using it in the various places

**Comment by nikriek** [comment]
> It seems the other DDL stats have the same syntax, e.g. `ALTER TABLE`. Could you fix them, too, in this PR?

Done. I fixed it for any occurrence of `tablePropertyList` in the SqlBase.g4 file.

**Comment by maropu** [comment]
LGTM

**Comment by maropu** [comment]
Ah, could you update the PR description, too?

**Comment by nikriek** [comment]
Done 👍

File: docs/sql-ref-syntax-ddl-create-table-hiveformat.md [hunk 2]
```diff
@@ -37,11 +37,11 @@ CREATE [ EXTERNAL ] TABLE [ IF NOT EXISTS ] table_identifier
     [ ROW FORMAT row_format ]
     [ STORED AS file_format ]
     [ LOCATION path ]
-    [ TBLPROPERTIES ( key1=val1, key2=val2, ... ) ]
+    [ TBLPROPERTIES ( key1 [=] val1, key2 [=] val2, ... ) ]
     [ AS select_statement ]
 
 row_format:    
-    : SERDE serde_class [ WITH SERDEPROPERTIES (k1=v1, k2=v2, ... ) ]
+    : SERDE serde_class [ WITH SERDEPROPERTIES (key1 [=] val1, key2 [=] val2, ...)  ]
```
Comment (huaxingao) [inline docs/sql-ref-syntax-ddl-create-table-hiveformat.md:44]: you accidentally added an extra space in between of `)` and `]`?

Comment (nikriek) [reply docs/sql-ref-syntax-ddl-create-table-hiveformat.md]: Thanks! I removed it.

File: docs/sql-ref-syntax-ddl-alter-table.md [hunk 3]
```diff
@@ -219,9 +219,9 @@ ALTER TABLE table_identifier [ partition_spec ] SET LOCATION 'new_location'
 
     Specifies the partition on which the property has to be set. Note that one can use a typed literal (e.g., date'2019-01-02') in the partition spec.
 
-    **Syntax:** `PARTITION ( partition_col_name  = partition_col_val [ , ... ] )`
+    **Syntax:** `PARTITION ( partition_col_name = partition_col_val [ , ... ] )`
 
-* **SERDEPROPERTIES ( key1 = val1, key2 = val2, ... )**
+* **SERDEPROPERTIES ( key1 [=] val1, key2 [=] val2, ... ) **
```
Comment (maropu) [inline docs/sql-ref-syntax-ddl-alter-table.md:224]: nit `) **` => `)**` (a unnecessary space?)

**Comment by maropu** [comment]
@nikriek Could you resolve the conflict? Also, could you fix the same issue in https://github.com/apache/spark/blame/master/docs/sql-ref-syntax-hive-format.md#L33, too?

**Pull request closed without merging**